types/beacon_types.go

```diff
@@ -231,16 +231,44 @@ func (n BeaconNetwork) EstimatedCurrentSlot() spec.Slot {
 
 // EstimatedSlotAtTime estimates slot at the given time
 func (n BeaconNetwork) EstimatedSlotAtTime(time int64) spec.Slot {
-	genesis := int64(n.MinGenesisTime())
-	if time < genesis {
+	// Sanitize time
+	if time < 0 {
 		return 0
 	}
-	return spec.Slot(uint64(time-genesis) / uint64(n.SlotDurationSec().Seconds()))
+
+	// Get delta time
+	genesis := n.MinGenesisTime()
+	deltaTime := uint64(time) - genesis
+
+	// Get slot duration
+	slotDurationInSeconds := int64(n.SlotDurationSec().Seconds())
+	// Sanitize slot duration
+	if slotDurationInSeconds <= 0 {
+		return spec.Slot(math.MaxUint64)
+	}
+
+	return spec.Slot(uint64(deltaTime) / uint64(slotDurationInSeconds))
 }
 
 func (n BeaconNetwork) EstimatedTimeAtSlot(slot spec.Slot) int64 {
-	d := int64(slot) * int64(n.SlotDurationSec().Seconds())
-	return int64(n.MinGenesisTime()) + d
+	// Get slot duration
+	slotDurationInSeconds := int64(n.SlotDurationSec().Seconds())
+	if slotDurationInSeconds < 0 {
+		return int64(math.MaxInt64)
+	}
+
+	// Get delta to add to genesis time
+	d := uint64(slot) * uint64(slotDurationInSeconds)
+
+	// Get genesis time
+	minGenesisTime := n.MinGenesisTime()
+
+	// Sanitize variables
+	if minGenesisTime > uint64(math.MaxInt64) || d > uint64(math.MaxInt64) {
+		return int64(math.MaxInt64)
+	}
+
+	return int64(minGenesisTime) + int64(d)
```

Comment:
converting from uint64 to int64 is considered safe by gosec?
A big number can turn negative I think (no that it will happen but you know)

Reply:
Notice that, before the conversion, sanity checks ensure it'll be converted properly. GoSec sees this. If I remove the checks, it will raise the error.

Reply:
I think... that if the sum of `int64(minGenesisTime) + int64(d)` will be larger than `math.MaxInt64` there will be an overflow... weird that go-sec didn't shout about that...

Anyhow, using some safe arithmetics here is an overkill since we both know that the above scenario will never happen and we don't want to have too many impl details in spec...
So if they shout I will try to talk to them about this

Reply:
By "they shout" you mean the implementation team?